### ISSUE TITLE
AWS S3 support

### ISSUE THREAD

**Issue by hirenaspl** [Elendev/ElendevRoxyFilemanBundle#7]
Hi Jonas,

Hope you are doing well.

Do you think we have any built-in support for AWS S3 storage?

If not then I understand that all we need is to create a "AwsS3FileSystem" class implementing FileSystemInterface class and then set its service id.

Please correct me if wrong.

Please help,
Hiren

**Comment by Elendev** [comment]
Hello Hiren,

There is no built-in support for AWS S3 storage for now, but it can be an nice feature to have.

We could use the Amazon PHP SDK ([https://aws.amazon.com/sdk-for-php](https://aws.amazon.com/sdk-for-php)), it provides support for the `s3://` stream [here](https://docs.aws.amazon.com/aws-sdk-php/v3/api/class-Aws.S3.StreamWrapper.html).

Adding AWS S3 SDK as a dependency and using the `StreamWrapper` might be enough to use S3 in conjunction with this bundle ?

**Comment by Elendev** [comment]
@hirenaspl let me know if using the stream wrapper `s3://` did work, if it did work I would gladly update the documentation.

**Comment by hirenaspl** [comment]
Hi Jonas,

Thanks for responding. Actually, we switched to https://github.com/servocoder/RichFilemanager and its server side connector https://github.com/servocoder/RichFilemanager-PHP as it has such support already. So actually, we haven't been able to proceed with above. But as per my understanding, the way you mentioned should work. Richfilemanager do have such thing already so in our case, it should work.

Our side solution was very urgent hence we switched to ready-made. I hope that's fine.

Thanks for all help and support,
Hiren
